src/query/service/src/interpreters/interpreter_table_create.rs

```diff
@@ -275,6 +282,52 @@ impl CreateTableInterpreter {
 
         Ok(req)
     }
+
+    async fn build_attach_request(&self, storage_prefix: &str) -> Result<CreateTableReq> {
+        let operator = self.ctx.get_data_operator()?.operator();
```

Comment:
sorry for interrupting (since pr is still draft), I might be wrong but seem here we need to construct a operator according to the storage parameters specified in the `attach table` stmt, so that later we can use this operator to read the snapshot from an external location.


for your reference:

- where the `storage_params` is extracted

https://github.com/datafuselabs/databend/blob/68181d1d6b70eb35e99ae873db51e784b103a413/src/query/sql/src/planner/binder/ddl/table.rs#L389-L412

- operator initialize with storage params

https://github.com/datafuselabs/databend/blob/071964285f91af59b3bb0a81715805aabf3b65db/src/query/storages/fuse/src/fuse_table.rs#L120-L122

Reply:
Thank you, I have realized this too